File: haskell/pangram/src/Pangram.hs
module Pangram
  ( isPangram
  ) where

import           Data.Char
import           Data.List.Split
import qualified Data.Set        as Set
import           Debug.Trace

isPangram :: String -> Bool
isPangram text
  | trace (show text) False = undefined
isPangram text = do
  let textLower = lowercase text
      letters = splitOn "" textLower
   in Set.size uniqueLetters letters == 26

lowercase :: String -> String
lowercase = map toLower

uniqueLetters :: String -> Set.Set
uniqueLetters letter = Set.fromList letters

File: solutions/csharp/pangram/1/Pangram.cs
public static class Pangram
{
    public static bool IsPangram(string input)
    {
        var uniqueLetters = new HashSet<char>();
        foreach (var c in input)
        {
            if (char.IsLetter(c))
            {
                uniqueLetters.Add(char.ToLower(c));
            }
        }
        return uniqueLetters.Count == 26;
    }
}
